wire/fromwire.c

```diff
@@ -165,6 +165,15 @@ bool fromwire_bool(const u8 **cursor, size_t *max)
 	return ret;
 }
 
+int fromwire_int(const u8 **cursor, size_t *max)
+{
+	int ret;
+
+	if (!fromwire(cursor, max, &ret, sizeof(ret)))
+		return 0;
+	return ret;
```

Comment:
This seems to be missing an endianness conversion, `be32_to_cpu` should do the trick.

Reply:
`be32` assumes 32-bit `int`s though.

Reply:
@ZmnSCPxj would you prefer [towire_s32()](e5fa911) over [towire_int()](732e790)?

I am still most inclined to "Use the existent `towire_u32()` for error codes (take just the first commit from this PR). We pass 400 and 401. Typecast to `int` when calling `command_fail()` because it takes an `int` argument."